pkg/assets/apiserver/asset_apiserver.go

```diff
@@ -131,7 +138,11 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 
 	s := &AssetServer{
 		GenericAPIServer: genericServer,
-		PublicURL:        c.ExtraConfig.PublicURL,
+
+		PublicURL: c.ExtraConfig.PublicURL,
+		// TODO figure out sttts envisions these being made available
+		BindAddress: c.ExtraConfig.Options.ServingInfo.BindAddress,
```

Comment:
note that we have two ways to pass them in: via a pre-created listener (for testing purposes especially) or with BindAddress+BindNetwork. SecureServingOptions.ApplyTo creates a listener in the second case. Here we should be long after this step and SecureServingInfo.Listener gives you both values. From looking at the commit here, I believe we should have either the listener here as well and pass it into AssetServer or we can "ask" the listener for both values.

Reply:
But in short: as we don't use the listener in the options for master bootstrapping, you code should be fine.

Reply:
> note that we have two ways to pass them in: via a pre-created listener (for testing purposes especially) or with BindAddress+BindNetwork. SecureServingOptions.ApplyTo creates a listener in the second case. Here we should be long after this step and SecureServingInfo.Listener gives you both values. From looking at the commit here, I believe we should have either the listener here as well and pass it into AssetServer or we can "ask" the listener for both values.

I want to spend zero more minutes on asset server, since it has already moved to its own repo.  This gets through the next week or two.